lumicks/pylake/kymotracker/kymotracker.py

```diff
@@ -368,7 +378,22 @@ def filter_tracks(tracks, minimum_length):
     minimum_length : int
         Minimum length for the track to be accepted.
     """
-    return KymoTrackGroup([track for track in tracks if len(track) >= minimum_length])
+
+    return KymoTrackGroup(
+        [
+            track._with_minimum_time(
+                max(
+                    # We can't unfilter tracks.
+                    track._minimum_observable_duration
+                    if track._minimum_observable_duration is not None
+                    else 0,
+                    (minimum_length - 1) * track._kymo.line_time_seconds,
```

Comment:
why the `-1`? If you filter with `minimum_length = 1` here you'd end up with a minimum observable time of `0` which is different behavior than what you have in `track_greedy`. Or even worse if for some reason you filter with `minimum_length=0` you get a weird timey whimey lower temporal bound

Reply:
We're filtering by points, not duration here. A minimum length of two points means a minimum duration of `dt` and so forth. The lowest bound on the duration for `N` points is `(N-1) * dt`.

Tracks coming from the tracker initialize with a minimum length of `dt` (the minimum length possible for a dwell to be a dwell) and therefore there cannot be timey wimey stuff (note the `max`).

There is a good reason to initialize with `dt` coming from the tracker since this is the minimally realizable duration. Any dwelltime analysis omits tracks with fewer than two points, since duration is a concept that makes no sense in this case.

I do think that given this comment, I suspect we'd be better off describing the parameter description as `Minimum number of points for the track to be accepted.`

Reply:
ok yeah, it is a bit to wrap your head around, but it makes sense